remote-messaging/remote-messaging-impl/src/main/java/com/duckduckgo/remote/messaging/impl/pixels/RemoteMessagingPixels.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.remote.messaging.impl.pixels
+
+import com.duckduckgo.app.statistics.pixels.Pixel
+import com.duckduckgo.app.statistics.pixels.Pixel.PixelType
+import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.remote.messaging.api.RemoteMessage
+import com.squareup.anvil.annotations.ContributesBinding
+import javax.inject.Inject
+
+interface RemoteMessagingPixels {
+
+    fun fireRemoteMessageShownPixel(remoteMessage: RemoteMessage)
+    fun fireRemoteMessageDismissedPixel(remoteMessage: RemoteMessage)
+    fun fireRemoteMessagePrimaryActionClickedPixel(remoteMessage: RemoteMessage)
+    fun fireRemoteMessageSecondaryActionClickedPixel(remoteMessage: RemoteMessage)
+    fun fireRemoteMessageActionClickedPixel(remoteMessage: RemoteMessage)
+    fun fireRemoteMessageSharedPixel(remoteMessage: Map<String, String>)
+}
+
+@ContributesBinding(AppScope::class)
+class RealRemoteMessagingPixels @Inject constructor(
+    private val pixel: Pixel,
+) : RemoteMessagingPixels {
+    override fun fireRemoteMessageShownPixel(remoteMessage: RemoteMessage) {
```

Comment:
There are 2 shown pixels, one unique and another one for everytime the RMF is displayed. It seems we are missing the shown aggregate.

Reply:
+1 let's have a look at that.

Reply:
sorted